### PR TITLE
Fix for issue #15320: Updates for SinonJS 2.x API...

### PR DESCRIPTION
**Pull request by rationull**
...including promise rejection/resolution from stubs.

Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://sinonjs.org/releases/v2.1.0/
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: sinon/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// Type definitions for Sinon 1.16.0
+// Type definitions for Sinon 2.1.0
 // Project: http://sinonjs.org/
-// Definitions by: William Sears <https://github.com/mrbigdog2u>
+// Definitions by: William Sears <https://github.com/mrbigdog2u>, Jonathan Little <https://github.com/rationull>
```
Comment (rationull) [inline sinon/index.d.ts]: Maybe adding attribution isn't actually appropriate for such a small change but I'm not sure what the standards are for this...

Comment (sandersn) [inline sinon/index.d.ts]: Do you want to be a reviewer on future changes to SinonJS? The original author only contributed the initial 3 commits (over 4 years ago), so I think it's fine to add more reviewers.

Comment (rationull) [inline sinon/index.d.ts]: Good point, I guess that's the main implication! I'm not even close to an authority on it and don't feel a need to approve future contributions. However I'm happy to be a reviewer to help out.

Comment (sandersn) [inline sinon/index.d.ts]: At this point @mrbigdog2u hasn't showed up after 2 days, so I would appreciate the help if you don't mind. 

I'll wait another day or so before merging to see if he does want to review it, though.

Comment (rationull) [inline sinon/index.d.ts]: Sure thing!

**Comment by andy-ms** [comment]
We've recently updated the repository structure.
All packages are now in a `types` subdirectory.
Please rebase this pull request onto the latest `master` branch.

**Comment by rationull** [comment]
@andy-ms Thanks for the heads-up, the PR is rebased now.

**Comment by rationull** [comment]
I did an actual rebase so the old commits and comments are lost .. status as of @sandersn last comment is we were waiting to see if @mrbigdog2u wanted to review this PR.

**Comment by sandersn** [comment]
5 days seems long enough to wait. I'll go ahead and merge.

**Comment by rationull** [comment]
Thanks @sandersn!

**Pull request merged**